packages/virtual-core/src/index.ts

```diff
@@ -79,7 +83,7 @@ export const observeElementRect = <T extends Element>(
     return () => {}
   }
 
-  const observer = new ResizeObserver((entries) => {
+  const observer = new targetWindow.ResizeObserver((entries) => {
```

Comment:
Also check if ResizeObserver is on targetWindow

Reply:
Here it should be guarded by type as well. (If you meant by checking `if (targetWindow.ResizeObserver)`)

Reply:
Updated :) could you give it one more round of review